odc/stats/_cli_save_tasks.py

```diff
@@ -99,18 +107,70 @@ def save_tasks(
       - output product config
 
     """
+    setup_logging()
+
+    import logging
     from datacube import Datacube
     from .tasks import SaveTasks
     from .model import DateTimeRange
 
-    filter = {}
-    if dataset_filter:
-        filter = json.loads(dataset_filter)
-
+    _log = logging.getLogger(__name__)
     if temporal_range is not None and year is not None:
         print("Can only supply one of --year or --temporal_range", file=sys.stderr)
         sys.exit(1)
 
+    if config is None:
+        config = {}
+
+    _cfg = {k: config.get(k) for k in CONFIG_ITEMS if config.get(k) is not None}
+
+    cfg_from_cli = {
+        k: v
+        for k, v in dict(
+            grid=grid,
+            frequency=frequency,
+            gqa=gqa,
+            products=products,
+            complevel=complevel,
+            dataset_filter=dataset_filter,
+        ).items()
+        if v is not None and v != ""
+    }
+
+    _log.info(f"Config overrides: {cfg_from_cli}")
+    _cfg.update(cfg_from_cli)
+    _log.info(f"Using config: {_cfg}")
+
+    gqa = _cfg.pop("gqa", None)
+    products = _cfg.pop("products", None)
+    dataset_filter = _cfg.pop("dataset_filter", None)
+
+    if products is None:
+        print("Input products has to be specified", file=sys.stderr)
+        sys.exit(1)
+
+    if _cfg.get("grid") is None:
+        print(
```

Comment:
Should print errors to `sys.stderr` like above. :)

I'm sure we don't do it everywhere, but I saw this one. :)

Reply:
note to myself, fix this in wofs pr